### ISSUE TITLE
MacOSX 10.10.2 / fzf 0.9.4 Memory Leak

### ISSUE THREAD

**Issue by jotham** [junegunn/fzf#185]
I've managed to consistently recreate a problem I've been having with fzf in OSX.

$ fzf --version
0.9.4
$ uname -a
Darwin 14.1.0 Darwin Kernel Version 14.1.0: Thu Feb 26 19:26:47 PST 2015; root:xnu-2782.10.73~1/RELEASE_X86_64 x86_64

If I open a terminal window and run (at a minimum) $(fzf) then close the terminal window (Terminal.app) I notice fzf doesn't exit and uses increasingly more and more memory (regularly like so http://i.imgur.com/syEIYXu.png).

Not sure how to debug this any further for you.  If I just kill the login shell from another terminal the problem doesn't occur, it seems to be limited to closing Terminal.app.

**Comment by junegunn** [comment]
Thanks for the report. That's strange. I'm not sure but it can be a bug of Terminal.app. If you try `$(sleep 10000)` and close the window, you'll notice that `sleep` process is not correctly terminated. :confused:

**Comment by jotham** [comment]
Hey,

Not sure this provides any more help.

If I fashion the following script:

```
sighandler() {
   printf 'Caught signal %d) SIG%s\n' "$1" "$(kill -l "$1")" >> /tmp/traps
}
for sig in {1..31}; do
   trap "sighandler $sig" "$sig"
done
$(sleep 10000)
```

Then in /tmp/traps I get a SIGHUP when I close the Terminal.app window, and as expected "sleep 10000" is still running as a process.

If I swap $(sleep 10000) with $(fzf) then fzf exits correctly.

Following that, I've created a simple bash script only containing $(fzf) and it exits cleanly as well.

My initial use of fzf was in a bash function, which I guess runs in the bash child of the login shell process. 
```
 | \-+= 02354 root login -pf jotham
 |   \-+= 02355 jotham -bash
```

Where as my test scripts are running in a bash child of the bash child of the login process.

```
| \-+= 02354 root login -pf jotham
 |   \-+= 02355 jotham -bash
 |     \-+= 02636 jotham -bash
```

So, I'm not sure why this is happening. It may not have anything to do with fzf.  For now I've wrapped my bash functions bodies in () to create another sub-shell and it seems to work without leaving an orphaned fzf process.

```
vide() {
   (
      src=~/Downloads
      name=$(find -E $src -iregex '.*(rar|mov|mp4|mkv|avi)' | fzf)
      [[ -f "$name" ]] && { open -na vlc "$name"; }
   )
}
```

Anyway, sorry for the rambling response.  Hopefully this is helpful.

**Comment by junegunn** [comment]
Thanks for the detailed report. I noticed that fzf goes into an infinite loop when that happens. Fixed the problem and released 0.9.8.
